satpy/readers/iasi_ng_l2_nc.py

```diff
@@ -0,0 +1,418 @@
+# Copyright (c) 2017-2023 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+
+"""IASI-NG L2 reader implementation.
+
+This reader supports reading all the products from the IASI-NG L2 processing
+level:
+
+* IASI-L2-TWV
+* IASI-L2-CLD
+* IASI-L2-GHG
+* IASI-L2-SFC
+* IASI-L2-O3
+* IASI-L2-CO
+"""
+
+import re
+
+import netCDF4
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from .netcdf_utils import NetCDF4FsspecFileHandler
+
+
+class IASINGL2NCFileHandler(NetCDF4FsspecFileHandler):
+    """Reader for IASI-NG L2 products in NetCDF format."""
+
+    def __init__(self, filename, filename_info, filetype_info, **kwargs):
+        """Initialize object."""
+        super().__init__(filename, filename_info, filetype_info, **kwargs)
+
+        self.sensors = {"iasi_ng"}
+
+        # List of datasets provided by this handler:
+        self.dataset_infos = None
+
+        # Description of the available variables:
+        self.variable_desc = {}
+
+        # Description of the available dimensions:
+        self.dimensions_desc = {}
+
+        # Ignored variable patterns:
+        patterns = self.filetype_info.get("ignored_patterns", [])
+        self.ignored_patterns = [re.compile(pstr) for pstr in patterns]
+
+        # dataset aliases:
+        self.dataset_aliases = self.filetype_info.get("dataset_aliases", {})
+
+        # Transform the aliases on regex patterns:
+        self.dataset_aliases = {
+            re.compile(key): val for key, val in self.dataset_aliases.items()
+        }
+
+        # broadcasts timestamps flag:
+        self.broadcast_timestamps = self.filetype_info.get("broadcast_timestamps", False)
```

Comment:
please don’t use comments on every line, good variables names are explanatory enough! and the variable names don’t need to be short…